### DISCUSSION TITLE
Dynamic vertex buffers

### DISCUSSION THREAD

**Discussion by Gavin-Williams** [amerkoleci/Vortice.Windows#422] [Q&A]
I have the following code...
```
        // update dynamic vertex buffer
        ReadOnlySpan<VertexP> vspan = stackalloc VertexP[]
        {
            new(new Vector3(quad.P1, 0)),
            new(new Vector3(quad.P2, 0)),
            new(new Vector3(quad.P4, 0)),
            new(new Vector3(quad.P3, 0)),
        };
        MappedSubresource vbMappedRes = dc.Map(VertexBufferDynamic, MapMode.WriteDiscard);
        Unsafe.Copy(vbMappedRes.DataPointer.ToPointer(), vspan);
        dc.Unmap(VertexBufferDynamic, 0);
```
I can't work out how to get the dynamic data into the buffer. I get the error

> The type ReadOnlySpan<VertexP> may not be used as a type argument.

How can I get this to work?

**Comment by amerkoleci** [comment]
Hi,
Its very simple indeed, just do:
```
vspan.CopyTo(new Span<VertexP>(mappedResource.DataPointer.ToPointer(), vspan.Length));
```

Tell me if it works?

**Reply by Gavin-Williams** [reply]
No overload takes 2 parameters. We have
```
CopyTo(Span<T> destination);
```

**Reply by Gavin-Williams** [reply]
this might work... (checking it now)
vspan.CopyTo(vbMappedRes.AsSpan<VertexP>(VertexBufferDynamic));

**Reply by amerkoleci** [reply]
Version 3.4.3-beta is coming (it adds SetData/GetData helpers to ID3D11Buffer), meanwhile take look here:

https://github.com/amerkoleci/Vortice.Windows/blob/main/src/Vortice.Direct3D11/ID3D11Buffer.cs#L10

**Reply by amerkoleci** [reply]
Here you have a complete example:
https://github.com/amerkoleci/Vortice.Windows.Samples/blob/main/src/Direct3D11/03_DrawQuad/DrawQuadApp.cs

**Comment by Gavin-Williams** [comment]
Here's the working code with the current release. Looking forward to the new release.
```
        // update dynamic vertex buffer
        ReadOnlySpan<VertexP> vspan =
        [
            new(new Vector3(quad.P1, 0)),
            new(new Vector3(quad.P2, 0)),
            new(new Vector3(quad.P4, 0)),
            new(new Vector3(quad.P3, 0)),
        ];
        MappedSubresource vbMappedRes = dc.Map(VertexBufferDynamic, MapMode.WriteDiscard);
        vspan.CopyTo(vbMappedRes.AsSpan<VertexP>(VertexBufferDynamic));
        dc.Unmap(VertexBufferDynamic, 0);
```

**Reply by amerkoleci** [reply]
3.4.3 beta is on nuget, grab it
